Fix #1884: Thoroughly test TextInputCaseSensitiveEqualsRuleClassifierProvider

alokbharti opened the pull request:
<!--
  - Thanks for submitting code to Oppia! Please fill out the following as part of
  - your pull request so we can review your code more easily.
  -->

## Explanation
Fixes #1884 , added test cases for TextInputCaseSensitiveEqualsRuleClassifierProvider

## Checklist
<!-- Please tick the relevant boxes by putting an "x" in them. -->
- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary auto-generated code from Android Studio.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR is made from a branch that is up-to-date with "develop".
- [x] The PR's branch is based on "develop" and not on any other branch.
- [ ] The PR is **assigned** to an appropriate reviewer in both the **Assignees** and the **Reviewers** sections.

alokbharti committed:
Added text cases for TextInputCaseSensitiveEqualsRuleClassifierProvider

alokbharti committed:
Resolved failed lint tests due to naming convention, file end line rule and wrong file name

alokbharti commented:
@anandwana001 @rt4914 PTAL

alokbharti committed:
Merge branch 'develop' into test-text_input_case_sensitive_equals_rule_classifier

rt4914 commented on domain/src/test/java/org/oppia/android/domain/classify/rules/textinput/TextInputCaseSensitiveEqualsRuleClassifierProviderTest.kt:
```diff
@@ -0,0 +1,125 @@
+    val matches =
+      inputCaseSensitiveEqualsRuleClassifier.matches(answer = STRING_VALUE_TEST_UPPERCASE, inputs = inputs) // ktlint-disable max-line-length
+
+    Truth.assertThat(matches).isTrue()
```
Update this to `assertThat(matches).isTrue()`

And directly import it.

Apply this in other test cases too.

rt4914 commented on domain/src/test/java/org/oppia/android/domain/classify/rules/textinput/TextInputCaseSensitiveEqualsRuleClassifierProviderTest.kt:
```diff
@@ -0,0 +1,125 @@
+class TextInputCaseSensitiveEqualsRuleClassifierProviderTest {
+  private val STRING_VALUE_TEST_UPPERCASE = createString(value = "TEST")
+  private val STRING_VALUE_TEST_LOWERCASE = createString(value = "test")
+  private val STRING_VALUE = createString(value = "string")
```
Instead of `string` write `random` and update it to
`STRING_VALUE_RANDOM`

rt4914 commented on domain/src/test/java/org/oppia/android/domain/classify/rules/textinput/TextInputCaseSensitiveEqualsRuleClassifierProviderTest.kt:
```diff
@@ -0,0 +1,125 @@
+    val inputs = mapOf("x" to STRING_VALUE_TEST_UPPERCASE)
+
+    val matches =
+      inputCaseSensitiveEqualsRuleClassifier.matches(answer = STRING_VALUE_TEST_UPPERCASE, inputs = inputs) // ktlint-disable max-line-length
```
Do not use `// ktlint-disable max-line-length`, we are updating other test files too to remove this.
You can do something like this:
```
    val matches =
      inputCaseSensitiveEqualsRuleClassifier.matches(
        answer = STRING_VALUE_TEST_UPPERCASE,
        inputs = inputs
      )
```

rt4914 replied:
Ditto elsewhere.

alokbharti replied:
Okay, will do that. Thanks

rt4914 reviewed (changes requested):
Nice work @alokbharti 

Please remove `TODO` from `TextInputCaseSensitiveEqualsRuleClassifierProvider.kt` file and update this PR as per comments.

alokbharti committed:
Modified TextInputCaseSensitiveEqualsRuleClassifierProviderTest

- Removed TODO statements
- Replaced Truth import with Truth.assertThat
- Replaced STRING_VALUE with STRING_VALUE_RANDOM
- Removed ktlint-disabling comments

rt4914 commented:
@alokbharti Procedure related to comments:
1. Until and unless all comments are resolved PR should not be merged.
2. Person who has started the comment thread, should resolve that comment.

So, in this case, I had started the comments and therefore they were suppose to be resolved by me. Can you open all those comments again?

Thanks.

rt4914 reviewed (changes requested):
@alokbharti Thanks. You have missed one comment:
https://github.com/oppia/oppia-android/pull/1961#pullrequestreview-505616263

alokbharti committed:
Removed TODO from TextInputCaseSensitiveEqualsRuleClassifierProvider class

alokbharti commented:
> @alokbharti Thanks. You have missed one comment:
> [#1961 (review)](https://github.com/oppia/oppia-android/pull/1961#pullrequestreview-505616263)

Committed the requested changes

rt4914 reviewed (approved):
LGTM, thanks
